### ISSUE TITLE
Lint for Min/Max If Statements

### ISSUE THREAD

**Issue by mmstick** [Manishearth/rust-clippy#1006]
Clippy should be able to detect instances where an if statement is being used to calculate the min or max of two or more values and recommend using the `cmp::min/max` functions instead. These two examples were taken from real-world code directly translated from C into Rust.

### Unnecessary
```rust
let mut x = if a < b { a } else { b };
if x > y { x = y; }
```

### Recommended
```rust
let x = cmp::min(cmp::min(a ,b), y);
```

### Unnecessary
```rust
let mut count = (sum + stride / 2) / stride;
if count < 1 { count = 1; }
if sum == 0 { count = 0; }
```

### Recommended
```rust
let count = if sum == 0 { 0 } else { cmp::max((sum + stride / 2) / stride, 1) };
```

**Comment by mcarton** [comment]
```rust
let x = cmp::min(cmp::min(a ,b), y);
```
could even be `[a, b, y].iter().min().unwrap()` which is better IMO.

**Comment by mmstick** [comment]
Although I don't know how I feel about it requring `unwrap()`. In what scenario would it return `None`?

**Comment by mcarton** [comment]
```rust
[].iter.min()
```

**Comment by shepmaster** [comment]
I'd disagree that the iterator version is better. It introduces the potential for failure (the `unwrap`) when the original one didn't have that potential. Chained `min` or `max` calls would be better.

**Comment by mcarton** [comment]
The failure case would be using `.iter().min()` on an empty array, it's not a problem here. Anyway, we can also suggest both and let the user choose.
